android/src/main/java/com/brentvatne/common/api/DRMProps.kt

```diff
@@ -0,0 +1,82 @@
+package com.brentvatne.common.api
+
+import com.brentvatne.common.toolbox.ReactBridgeUtils.safeGetArray
+import com.brentvatne.common.toolbox.ReactBridgeUtils.safeGetString
+import com.facebook.react.bridge.ReadableMap
+
+/**
+ * Class representing DRM props for host.
+ * Only generic code here, no reference to the player.
+ */
+class DRMProps {
+    /**
+     * string version of configured UUID for drm prop
+     */
+    var drmType: String? = null
+
+    /**
+     * DRM license server to be used
+     */
+    var drmLicenseServer: String? = null
+
+    /**
+     * DRM Http Header to access to license server
+     */
+    var drmLicenseHeader: Array<String> = emptyArray<String>()
+
+    /** return true if this and src are equals  */
+    override fun equals(other: Any?): Boolean {
+        if (other == null || other !is DRMProps) return false
+        val seemsEqual = (
+            drmType == other.drmType &&
+                drmLicenseServer == other.drmLicenseServer &&
+                drmLicenseHeader.size == other.drmLicenseHeader.size
```

Comment:
In DRM, headers are important in most cases, so it seems better to use something like contentDeepEquals to make a clear comparison rather than a size comparison.

Reply:
I meant to recommend using the Array.contentDeepEquals function! ([contentDeepEquals](https://github.com/JetBrains/kotlin/blob/v1.7.0/libraries/stdlib/src/kotlin/collections/Arrays.kt#L75) internally use size comparison)

Reply:
OK, clear, I patched it, thank you